File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) return false;

        map<char, int> map1, map2;

        // Count characters in both strings
        for (int i = 0; i < s.length(); i++) {
            map1[s[i]]++;
            map2[t[i]]++;
        }

        // Compare frequency of each character
        for (char c : s) {
            if (map1[c] != map2[c]) return false;
        }

        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) return false;
        int n = s.length();


        int[] arr = new int[26];

        for(int i = 0; i < n; i++) {
            arr[s.charAt(i) - 'a'] += 1;
            arr[t.charAt(i) - 'a'] -= 1;
        }

        for(int i = 0; i < 26; i++) {
            if(arr[i] != 0) return false;
        }

        return true;
    }
}